src/amo/components/AddonReview.js

```diff
@@ -47,18 +48,34 @@ export class AddonReviewBase extends React.Component {
     const addonSlug = this.props.review.addonSlug;
     const apiState = this.props.apiState;
 
+    const newReviewParams = {
+      body: reviewBody,
+    };
+
+    const updatedReview = {
+      ...this.props.review,
+      ...newReviewParams,
+    };
+
     const params = {
       addonSlug,
       apiState,
-      body: reviewBody,
       errorHandler: this.props.errorHandler,
       reviewId: this.props.review.id,
+      ...newReviewParams,
     };
     // TODO: render a progress indicator in the UI.
     // https://github.com/mozilla/addons-frontend/issues/1156
+
+    // First, put the new review in state so that the
+    // component doesn't re-render with old data while
+    // our API request is in the air.
+    this.props.setDenormalizedReview(updatedReview);
+
+    // Next, update the review for real with an API call.
```

Comment:
Again, a bit nitpicky, but "update the review for real" might be better written as "update the review on the server with an API call"